src/lib.rs

```diff
@@ -413,6 +417,102 @@ impl Contract {
             .collect()
     }
 
+    pub fn get_addon(&self, id: AddOnId) -> Option<AddOn> {
+        self.available_addons.get(&id)
+    }
+
+    pub fn get_available_addons(&self) -> Vec<AddOn> {
+        self.available_addons.iter().map(|(_id, add_on)| add_on).collect()
+    }
+
+    // Only the contract admin and DevHub moderators
+    pub fn create_new_addon(&mut self, addon: AddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can create new add-ons");
+        }
+        if self.get_addon(addon.id.to_owned()).is_some() {
+            panic!("Add-on with this id already exists");
+        }
+        addon.validate();
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    // ONLY FOR TESTING
+    pub fn delete_addon(&mut self, id: AddOnId) {
+        // Also delete from communities
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can delete add-ons");
+        }
+        let addon = self
+            .get_addon(id.clone())
+            .expect(&format!("Add-on with id `{}` does not exist", id))
+            .clone();
+
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn update_addon(&mut self, input: AddOn) {
```

Comment:
`input` is really not a great name, at least, let's use `addon`:

```suggestion
    pub fn update_addon(&mut self, addon: AddOn) {
```